partiql-lang/src/main/kotlin/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -129,16 +127,28 @@ private typealias ThunkEnvValue<T> = ThunkValue<Environment, T>
  * @param compileOptions Various options that effect how the source code is compiled.
  */
 internal class EvaluatingCompiler(
-    private val functions: Map<String, ExprFunction>,
+    private val functions: List<ExprFunction>,
     private val customTypedOpParameters: Map<String, TypedOpParameter>,
     private val procedures: Map<String, StoredProcedure>,
     private val compileOptions: CompileOptions = CompileOptions.standard()
 ) {
+    @Deprecated(
+        "Functions should be a list. Use EvaluatingCompiler with List<ExprFunction> instead.",
+        level = DeprecationLevel.WARNING
+    )
+    constructor(
+        functions: Map<String, ExprFunction>,
+        customTypedOpParameters: Map<String, TypedOpParameter>,
+        procedures: Map<String, StoredProcedure>,
+        compileOptions: CompileOptions = CompileOptions.standard()
+    ) : this(functions.values.toList(), customTypedOpParameters, procedures, compileOptions)
```

Comment:
Really if we want a list of functions internally, we could just use the `map.values` iterable and not have to add all of these deprecations and duplications. It's not ideal, but it does simplify things.

But again, using the list is different because we'll be using function names from the signature rather than the names in the given map (which may differ!)